### PR TITLE
DOC: special: complete the docstrings of Chi-square functions

### PR DESCRIPTION
**Pull request by person142**
#### Reference issue
None.

#### What does this implement/fix?
Add:

- parameters/returns sections
- see also sections
- examples
- references

for `chdtr`, `chdtrc`, `chdtri`, and `chdtriv`.

#### Additional information
None.

### CODE REVIEW

**Comment by person142** [comment]
CI failures appear to be unrelated.

File: scipy/special/add_newdocs.py [hunk 1]
```diff
@@ -1388,46 +1388,212 @@ def add_newdoc(name, doc):
...
+        Degrees of freedom.
+    p : array_like
+        Probability.
+    out : ndarray, optional
+        Optional output array for the function results.
+
+    Returns
+    -------
+    x : scalar or ndarray
+        Value so that he probability a Chi square random variable
```
Comment (tylerjereddy) [inline scipy/special/add_newdocs.py]: typo `he` -> `the`, maybe just fix that in a subsequent PR when you get a chance

**Review by tylerjereddy — APPROVED** [review]
The docstrings are clearly superior to the bare bones versions that were present before & CI issues are indeed unrelated, thanks Josh.

**Comment by mdhaber** [comment]
Thanks @person142! I noticed you've never added your name to THANKS.txt, so if you'd like to, please make an entry here.

**Pull request merged**